### PR TITLE
feat(stdlibs): add package `html`

### PR DESCRIPTION
**Pull request by Villaquiranm**
related to #1267 
- Implement html golang package.
- Added a silly implementation of strings.Replacer:
Maybe we should optimize it ? I did not wanted to do something complex as this is not the main point of this pull request 

<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

File: gnovm/stdlibs/strings/replace.gno
Comment (thehowl) [inline gnovm/stdlibs/strings/replace.gno:1]: I suggest you do another PR which adds support for strings.Replacer, and then re-base this one on top of that.

- We should not introduce more "temporary" code in the standard library, when we are gearing up towards mainnet.
- This code has the fault of being "iterative" on the replacements; ie. `NewReplacer("a", "bb", "b", "cc").Replace("a")` returns `cccc` rather than `bb`.

**Review by thehowl — COMMENTED** [review]
I see other changes concern removing sync.Once.

LGTM after we fixup strings.Replacer in another PR.

File: gnovm/stdlibs/strings/replace.gno
Comment (Villaquiranm) [reply gnovm/stdlibs/strings/replace.gno:1]: You're right thanks for the review :)

**Review by moul — APPROVED** [review]
I think we will want to remove this, but it looks good to add it while we're still exploring. lgtm.

File: gnovm/stdlibs/strings/replace.gno
Comment (Villaquiranm) [reply gnovm/stdlibs/strings/replace.gno:1]: @thehowl PR for string Replacer here :)
thanks 
#2816

File: gnovm/stdlibs/html/entity.gno [hunk 1]
```diff
@@ -0,0 +1,2263 @@
...
+//
+// Note that the HTML5 list is larger than the HTML4 list at
+// http://www.w3.org/TR/html4/sgml/entities.html
+var entity map[string]rune
+
+// HTML entities that are two unicode codepoints.
+var entity2 map[string][2]rune
+
+// populateMapsOnce guards calling populateMaps.
+var populateMapsOnce bool
```
Comment (petar-dambovaliev) [inline gnovm/stdlibs/html/entity.gno:22]: Can we remove this bool and check by the map size?

Comment (petar-dambovaliev) [reply gnovm/stdlibs/html/entity.gno:22]: Or maybe you can populate these maps directly in the declaration? It is guaranteed to be initialized only once.
Why is `populateMaps` necessary ?

Comment (thehowl) [reply gnovm/stdlibs/html/entity.gno:22]: you're right; this is the kind of optimization that somewhat makes sense for go but i think it's fine if we discard for gno.

so let's just have these in the declarations directly.

Comment (Villaquiranm) [reply gnovm/stdlibs/html/entity.gno]: Sure thanks @petar-dambovaliev @thehowl  for the review indeed it is not required 👍 
I removed the function and just initialize directly both tables on a5f4a1b4f6f5583200a3ff0ce9cd1f7274173513

**Comment by Villaquiranm** [comment]
> I see other changes concern removing sync.Once.
> 
> LGTM after we fixup strings.Replacer in another PR.

@thehowl rebased :)

**Review by thehowl — APPROVED** [review]
LGTM :clap:
